src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -499,8 +523,19 @@ def recordInCouch(self, jobs, newstate, oldstate, updatesummary=False):
                                 jobSummary[prop] = jobSummary[prop] if jobSummary[prop] else currentJobDoc.get(prop, [])
                         except CouchNotFoundError:
                             pass
-                    if countDocs >= self.fwjrdatabase.getQueueSize():
-                        self.jsumdatabase.commit()
+
+                    if self.jsumdatabase.getQueueSize() >= self.maxBulkCommit:
+                        try:
+                            # TODO FIXME: https://github.com/dmwm/WMCore/issues/11576
+                            self.jsumdatabase.commit()
+                        except CouchRequestTooLargeError as exc:
+                            msg = "Failed to commit bulk of job summary documents to CouchDB."
+                            msg += f" Details: {str(exc)}"
+                            logging.warning(msg)
+                            shrinkLargeSummary(self.jsumdatabase, self.fwjrLimitSize)
```

Comment:
Ok, Let me see if I read it correctly: If we are in this situation, calling the `shringLargeSummary` function here is going to trigger a full scan of the local database and shrink only, the documents which are exceeding the configured size by cutting the fields that pass over some threshold? is My understanding correct?